File: src/lib-common/is-identifier.js
function is_identifier(x)
{
 if(!is_str(x))
  return false

 if(is_empty(x))
  return false
  
 const s=front(x)
 
 if(!is_alpha(s))
  return false
  
 for(const v of x)
 {
  if(!is_alnum(v))
   return false
 }
 
 return true
}

File: src/lib-common/is/is-identifier.cs
fn is_identifier x
 if not is_str x
  ret false

 if is_empty x
  ret false

 let s front x

 if same s "_"
 elseif is_alpha s
 else
  ret false

 for x
  if not is_alnum v
   ret false
 end

 ret true
end
